drivers/sensor/paw3212/paw3212.c

```diff
@@ -257,6 +282,114 @@ static int reg_write(struct paw3212_data *dev_data, u8_t reg, u8_t val)
 	return 0;
 }
 
+static int update_cpi(struct paw3212_data *dev_data, u32_t cpi)
+{
+	int err;
+
+	if ((cpi > PAW3212_CPI_MAX) || (cpi < PAW3212_CPI_MIN)) {
+		LOG_ERR("CPI %" PRIu32 " out of range", cpi);
+		return -EINVAL;
+	}
+
+	u8_t regval = cpi / PAW3212_CPI_STEP;
+
+	LOG_INF("Set CPI: %u (requested: %u, reg:0x%" PRIx8 ")",
+		regval * PAW3212_CPI_STEP, cpi, regval);
+
+	err = reg_write(dev_data, PAW3212_REG_WRITE_PROTECT, PAW3212_WPMAGIC);
+	if (err) {
+		LOG_ERR("Cannot disable write protect");
+		return err;
+	}
+
+	err = reg_write(dev_data, PAW3212_REG_CPI_X, regval);
+	if (err) {
+		LOG_ERR("Failed to change x CPI");
+		return err;
+	}
+
+	err = reg_write(dev_data, PAW3212_REG_CPI_Y, regval);
+	if (err) {
+		LOG_ERR("Failed to change y CPI");
+		return err;
+	}
+
+	err = reg_write(dev_data, PAW3212_REG_WRITE_PROTECT, 0);
+	if (err) {
+		LOG_ERR("Cannot enable write protect");
+	}
+
+	return err;
+}
+
+static int update_sleep_timeout(struct paw3212_data *dev_data, u8_t reg_addr,
+				u32_t timeout_ms)
+{
+	u32_t timeout_step_ms;
+
+	switch (reg_addr) {
+	case PAW3212_REG_SLEEP1:
+		timeout_step_ms = 32;
+		break;
+
+	case PAW3212_REG_SLEEP2:
+	case PAW3212_REG_SLEEP3:
+		timeout_step_ms = 20480;
+		break;
+
+	default:
+		LOG_ERR("Not supported");
+		return -ENOTSUP;
+	}
+
+	u32_t etm = timeout_ms / timeout_step_ms - 1;
+
+	if ((etm < PAW3212_ETM_MIN) || (etm > PAW3212_ETM_MAX)) {
+		LOG_WRN("Sleep timeout %" PRIu32 " out of range", timeout_ms);
+		return -EINVAL;
+	}
+
+	LOG_INF("Set sleep%d timeout: %u (requested: %u, reg:0x%" PRIx8 ")",
+		reg_addr - PAW3212_REG_SLEEP1 + 1,
+		(etm + 1) * timeout_step_ms,
+		timeout_ms,
+		etm);
+
+	int err;
+
+	err = reg_write(dev_data, PAW3212_REG_WRITE_PROTECT, PAW3212_WPMAGIC);
+	if (err) {
+		LOG_ERR("Cannot disable write protect");
+		return err;
+	}
+
+	u8_t regval;
+
+	err = reg_read(dev_data, reg_addr, &regval);
+	if (err) {
+		LOG_ERR("Failed to read sleep register");
+		return err;
+	}
+
+	__ASSERT_NO_MSG((etm & PAW3212_ETM_MASK) == etm);
+
+	regval &= ~PAW3212_ETM_MASK;
+	regval |= etm;
```

Comment:
`(etm << PAW3212_ETM_POS)`, for consistency.

Reply:
Right! Fixed